controllers/auth.js

```diff
@@ -9,6 +9,91 @@ const {
   USER_DOES_NOT_EXIST_ERROR,
 } = require("../constants/errorMessages");
 
+const googleAuthLogin = (req, res, next) => {
+  const { redirectURL, dev } = req.query;
+  if (dev === "true") {
+    return passport.authenticate("google", {
+      scope: ["email"],
+      state: redirectURL,
+    })(req, res, next);
+  } else {
+    return res.boom.unauthorized("User cannot be authenticated");
+  }
+};
+
+async function handleGoogleLogin(req, res, user, authRedirectionUrl) {
+  const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl"));
+  try {
+    if (!user.emails || user.emails.length === 0) {
+      throw new Error("Email not found in user data");
+    }
+    const userData = {
+      email: user.emails[0].value,
+      created_at: Date.now(),
+      updated_at: null,
```

Comment:
While Google typically provides validated emails, as a best practice we should still validate the email format from our end. This provides an additional security layer. Let's say for some reason `user.emails[0]` is `undefined`, this could throw an error. Also, Google can return multiple emails, and we should ideally check for the primary email.

Reply:
I’ve added a basic check for primary emails also now to ensure that the user’s email data exists and is properly verified. The code verifies that:
The user.emails array is not empty, and
The first verified email is selected.
While Google already verifies emails on their end, this additional check ensures that we're handling any unexpected cases (e.g., missing or unverified emails) gracefully. Typically, the first verified email should be the primary one, and this is the one we use.